### DISCUSSION TITLE
How to include a poster for a video

### DISCUSSION THREAD

**Discussion by maucejo** [quarto-dev/quarto-cli#9381] [Q&A] [answered]
### Description

Dear all,

After looking at the documentation, I don't find a way to include a poster to an embedded video.
Do you know how to do this ? It could be nice to use a syntax like `![](myvideo){poster=poster.png}`.

Than you for your help

**Comment by mcanouil** [comment]
~~If this syntax does not work, it means it is not supported by Pandoc.~~ Note that you might be missing the quotes in your attribute.
You can use raw HTML in raw blocks as described in "Markdown Basics".

Edit: the syntax works as expected as long as you respect the syntax.

```md
![](video.mp4){poster="image.png"}
```

**Reply by maucejo** [reply]
Thanks it works @mcanouil with a slight bug however. Indeed after inspection of the resulting html output, it comes that if the video and the poster is in the `images` folder (created in the quarto book project folder), I have to enter `![](/images/video.mp4){poster=".\images/image.png"}` and to copy the poster image in the `_book/images/` folder (the image is not directly copied).

Furthermore, I tried to mute the video using `![](/images/video.mp4){poster=".\images/image.png" controls muted}`. However, this command is not properly translated in the html output. Of course, I can use a raw HTML block as suggested, and it works.

**Reply by mcanouil** [reply]
Why do you have a backslash in the poster attribute? `poster=".\images/image.png"` is wrong.
This might be the main issue here.
When not discovered automatically by Quarto, simply add the resource in your configuration file: https://quarto.org/docs/websites/website-tools.html#site-resources

Note that your "example" is quite unclear about what the structure of your setup is.

**Reply by maucejo** [reply]
You're right, I have forgotten to describe my setup.

I am working on Windows 11. I have created a book project in VSCode using the Quarto book template.

Regarding the backslash, I haven't added it initially but it is what is included in the html output by the parser when I include a video using `![](/images/video.mp4)`. When I include the poster as `![](/images/video.mp4){poster="/images/poster.png"}`, the poster is not displayed.  It works when replacing the first slash by the backslash.

I will continue to investigate.

**Reply by mcanouil** [reply, marked as answer]
The path for "poster" is relative to the root of the website, but here you set it as absolute to the root of the file system.
Quarto does not handle all paths.
The proper syntax would be:

```md
![](/images/video.mp4){poster="./images/poster.png"}
```

Eventually with the following in your `_quarto.yml`

```yaml
project:
  resources:
    - /images/poster.png
```
